// node-mysql/main.js
var path = require("path"),
    fs = require("fs"),
    mysql = require("mysql");

var wwwroot = process.cwd();
    mysqlConf = JSON.parse(fs.readFileSync(path.join(wwwroot, "/node_modules/node-mysql/mysql.conf")));

/**************************** ClientMysql  start ***********************/
function clientMysql(sql, callback) {

    var connection = mysql.createConnection({
        host: mysqlConf.host, //主机
        user: mysqlConf.user, //MySQL认证用户名
        password: mysqlConf.password, //MySQL认证用户密码
        port: mysqlConf.port, //端口号
        database: mysqlConf.database
    });
    connection.connect(function(err) {
        if (err) {
            callback(true, "[connect err] " + err)
            return {
                success: false,
                result: err
            };
        }
    });
    connection.query(sql, callback);
    connection.end();
}

exports.clientMysql = clientMysql;

/**************************** ClientMysql  end ***********************/



/**************************** INSERTMysql  start ***********************/

function InsertMysql(tableName, values, callback) {

    try {
        var valueStr = "";
        values.forEach(function(oneValue, index) {
            if (oneValue == null) {
                valueStr += oneValue + ",";
            } else {
                valueStr += "\'" + oneValue + "\',";
            }

        })
        valueStr = valueStr.substring(0, valueStr.length - 1);
        var Sql = "INSERT INTO " + tableName + " VALUES(" + valueStr + ")";
        clientMysql(Sql, function(err, result) {
            callback && callback(err, result)
        })
    } catch (e) {
        return {
            success: false,
            result: "wrong arguments"
        }
    }

}

exports.InsertMysql = InsertMysql;

/**************************** INSERTMysql  end ***********************/


/**************************** DELETEMysql  start ***********************/

function DELETEMysql(tableName, field, value, callback) {

    var Sql = 'DELETE FROM ' + tableName + ' WHERE ' + field + ' = "' + value + '"';

    clientMysql(Sql, function(err, result) {
        callback && callback(err, result);
    })
}

exports.DELETEMysql = DELETEMysql;

/**************************** DELETEMysql  end ***********************/
